File: examples/searchbookings.clj
; Room Search example in Clojure
; Owain Kenway

; Make sure we have the right number of command line arguments.
(if (= (count *command-line-args*) 2)
  (def token (first *command-line-args*))
  ((.println System/err "Run with API key and date as arguments.")(System/exit 1))
)

; Import necessary classes.
(import uclapi.UCLApiConnection)
(import uclapi.Booking)
(import java.util.Hashtable)

; Put search parameters into a Hashtable.
(def params (Hashtable.))
(.put params "date" (second *command-line-args*))

; Do our search and print results.
(println (clojure.string/join "\n" (Booking/searchAPI (UCLApiConnection. token) params)))

File: examples/searchbookings.py
# UCL API example in Jython
# This example queries the roombookings/rooms endpoint and prints results.

# Owain Kenway

import uclapi.UCLApiConnection
import uclapi.Booking
import java.util.Hashtable

import sys

# Get our key + query.
if len(sys.argv) != 3:
	sys.exit("Call with key and date in [YYYYMMDD] as arguments")

key = sys.argv[1]
params = java.util.Hashtable()
params.put("date", sys.argv[2])

# Make our connection and get our results.
conn = uclapi.UCLApiConnection(key)
results = uclapi.Booking.searchAPI(conn, params)

# Print them out.
for a in results:
	print(a.toString())
